libmscore/system.cpp

```diff
@@ -1165,33 +1169,12 @@ MeasureBase* System::nextMeasure(const MeasureBase* m) const
 
 //---------------------------------------------------------
 //   scanElements
-//    collect all visible elements
 //---------------------------------------------------------
 
 void System::scanElements(void* data, void (* func)(void*, Element*), bool all)
 {
-    if (vbox()) {
-        return;
-    }
-    for (Bracket* b : _brackets) {
-        func(data, b);
-    }
-
-    if (_systemDividerLeft) {
-        func(data, _systemDividerLeft);
-    }
-    if (_systemDividerRight) {
-        func(data, _systemDividerRight);
-    }
-
-    int idx = 0;
-    for (const SysStaff* st : _staves) {
-        if (all || st->show()) {
-            for (InstrumentName* t : st->instrumentNames) {
-                func(data, t);
-            }
-        }
-        ++idx;
+    for (ScoreElement* child : *this) {
+        child->scanElements(data, func, all);
     }
     for (SpannerSegment* ss : _spannerSegments) {
```

Comment:
I know that the code below here is not yours, but it could still use the nicer syntax I suggested [here](https://github.com/musescore/MuseScore/pull/6274#discussion_r454745688). (Though I actually think it would be better to put it in `SpannerSegment::scanElements`.)

Reply:
Yep, changed the syntax and moved to `SpannerSegment`.